aion/lib/aion/room_channel/room.ex

```diff
@@ -73,15 +84,31 @@ defmodule Aion.RoomChannel.Room do
 
   @spec get_current_question(__MODULE__.t) :: Question.t
   def get_current_question(room) do
-    room.question
+    room.current_question
   end
 
-  @spec get_new_question_with_answers(integer) :: %{question: Question.t, answers: list Answer.t}
-  defp get_new_question_with_answers(room_id) do
-      question = Question.get_random_question(room_id)
-      answers = Answer.get_answers(question.id)
-      Logger.debug fn -> "Answers: #{inspect(Enum.map(answers, fn answer -> answer.content end))}" end
+  @spec get_new_question_with_answers((list Question.t), integer) ::  new_question_set
+  defp get_new_question_with_answers(questions, room_id) do
+    case questions do
+      [next_question | remaining_questions] ->
+        answers = Answer.get_answers(next_question.id)
+        Logger.debug fn -> "Answers: #{inspect(Enum.map(answers, fn answer -> answer.content end))}" end
+
+        remaining_questions = if Enum.empty?(remaining_questions) do
```

Comment:
If you want to format it this way would be much better if you do:
```
remaining_questions =
  if Enum.empty?(remaining_questions) do
    fetch_questions(room_id)
  else
    remaining_questions
  end
```

Reply:
I think it would actually be better to pattern match here, since `Enum.empty?` does not imply what the data structure type actually is (could be both a map or a list). Using pattern matching you are explicit about what the remaining questions are and therefore make it more legible. It would go as follows:
```
remaining_questions = 
     case remaining_questions do
          [] -> ...
          something -> something
     end
```
Also, it's just a nit, but maybe it would be cool to think of another name for the new variable (x = fun(x) isn't that easy to read. If you are changing the x, maybe there's a better naming for the new x or for the old x?). 

You can always pattern much better, like:

```
case questions do
     [next_question] -> 
     [next_question | remaining_questions] ->
end
``` 
^ it wouldn't force you to run additional ifs etc.

Reply:
I like this approach @mrapacz 👍